drizzlepac/hlautils/se_source_generation.py

```diff
@@ -387,46 +412,119 @@ def write_catalog(seg_cat, catalog_filename, product="tdp"):
+
+        # Add a description for columns which map to SExtractor catalog columns
+        seg_table["xcentroid"].description = "SExtractor Column x_image "
+        seg_table["ycentroid"].description = "SExtractor Column y_image "
+        seg_table["background_at_centroid"].description = "SExtractor Column background"
+        seg_table["source_sum"].description = "SExtractor Column flux_iso"
+        seg_table["source_sum_err"].description = "SExtractor Column fluxerr_iso"
+        # FIX: is mapping to _image or _world?  _image
+        seg_table["cxx"].description = "SExtractor Column cxx_image, ellipse parameter"
+        seg_table["cyy"].description = "SExtractor Column cyy_image, ellipse parameter"
+        seg_table["cxy"].description = "SExtractor Column cxy_image, ellipse parameter"
+        # FIX: is the mapping to _image or _world?
+        seg_table["covar_sigx2"].description = "SExtractor Column x2_image, (0,0) element of covariance matrix"
+        seg_table["covar_sigy2"].description = "SExtractor Column y2_image, (1,1) element of covariance matrix"
+        seg_table["covar_sigxy"].description = "SExtractor Column xy_image, (0,1) and (1,0) elements of covariance matrix"
+
+        seg_table["xmin"].description = "SExtractor Column xmin_image"
+        seg_table["xmax"].description = "SExtractor Column xmax_image"
+        seg_table["ymin"].description = "SExtractor Column ymin_image"
+        seg_table["ymin"].description = "SExtractor Column ymax_image"
+
+        # Write out the official filter detection product source catalog
+        seg_table["xcentroid"].info.format = ".10f"
+        seg_table["ycentroid"].info.format = ".10f"
         seg_table["RA_icrs"].info.format = ".10f"
         seg_table["Dec_icrs"].info.format = ".10f"
-        seg_table["RA_icrs"].unit = u.deg
-        seg_table["Dec_icrs"].unit =  u.deg
         print("seg_table (filter): {}".format(seg_table))
 
         seg_table.write(catalog_filename, format="ascii.ecsv")
         log.info("Wrote filter source catalog: {}".format(catalog_filename))
 
+def get_header_data(imghdu):
+
+    keyword_dict = {}
+
+    # Get the HSTWCS object from the primary header
+    keyword_dict["proposal_id"] = imghdu[0].header["PROPOSID"]
+    keyword_dict["image_file_name"] = imghdu[0].header['FILENAME']
+    keyword_dict["target_name"] = imghdu[0].header["TARGNAME"] 
+    keyword_dict["date_obs"] = imghdu[0].header["DATE-OBS"]
+    keyword_dict["instrument"] = imghdu[0].header["INSTRUME"]
+    keyword_dict["detector"] = imghdu[0].header["DETECTOR"]
+    keyword_dict["target_ra"] = imghdu[0].header["RA_TARG"]
+    keyword_dict["target_dec"] = imghdu[0].header["DEC_TARG"]
+    keyword_dict["expo_start"] = imghdu[0].header["EXPSTART"]
+    keyword_dict["expo_time"] = imghdu[0].header["EXPTIME"]
+    keyword_dict["ccd_gain"] = imghdu[0].header["CCDGAIN"]
+    keyword_dict["filter_tdp"] = imghdu[0].header["FILTER"]
+    keyword_dict["filter1"] = imghdu[0].header["FILTER1"]
```

Comment:
You will want logic to guard against using this on WFC3 data which does NOT have a FILTER1 or FILTER2 keyword which would cause this line to throw an exception (or at least I would expect it to).

Reply:
Yes, I keep forgetting about this.